pkg/ctl/get/iamserviceaccount.go

```diff
@@ -124,8 +84,9 @@ func doGetIAMServiceAccount(cmd *cmdutils.Cmd, serviceAccount *api.ClusterIAMSer
 	var obj interface{}
 	if params.output == "table" {
 		addIAMServiceAccountSummaryTableColumns(printer.(*printers.TablePrinter))
-		obj = cfg.IAM.ServiceAccounts
+		obj = serviceAccounts
 	} else {
+		cfg.IAM.ServiceAccounts = serviceAccounts
 		obj = cfg
 	}
 	return printer.PrintObjWithKind("iamserviceaccounts", obj, os.Stdout)
```

Comment:
Because we have recently been caught out on this twice now, could you please confirm that when `get ... -o json` or `get ... -o yaml` returns no service accounts the thing we output is an empty array, not null or an error.

Reply:
Good spot! Fixed it so it now returns:
```
eksctl get iamserviceaccounts --cluster jk
[ℹ]  eksctl version 0.38.0-dev+64cca6cf.2021-02-01T12:07:42Z
[ℹ]  using region us-west-2
No iamserviceaccounts found

eksctl get iamserviceaccounts --cluster jk -o json
[]

eksctl get iamserviceaccounts --cluster jk -o yaml
[]
```

I don't know why empty yaml is `[]`, it does work correctly when an account exists e.g.:
```
eksctl get iamserviceaccounts --cluster jk -o yaml
- metadata:
    name: fooo
    namespace: baz
  status:
    roleARN: arn:aws:iam::<redacted>:role/eksctl-jk-addon-iamserviceaccount-baz-fooo-Role1-11XZWGHLZ27DH
  wellKnownPolicies:
    autoScaler: false
    awsLoadBalancerController: false
    certManager: false
    externalDNS: false
    imageBuilder: false
```